File: frontend/src/dataTypes.js
const dataTypeMappings = {
  object: "Text",
  int64: "Integer (64-bit)",
  int32: "Integer (32-bit)",
  int16: "Integer (16-bit)",
  int8: "Integer (8-bit)",
  float64: "Floating Point (64-bit)",
  float32: "Floating Point (32-bit)",
  "datetime64[ns]": "Date & Time",
  "timedelta64[ns]": "Time Duration",
  bool: "Boolean (True/False)",
  category: "Categorical",
  complex: "Complex Number",
};

export default dataTypeMappings;

File: backend/data_cleanser/data_types.py
class DataTypes:
    """
    Constants representing different data types.
    """
    OBJECT = 'object'
    INTEGER = 'int64'
    INT64 = 'int64'
    INT32 = 'int32'
    INT16 = 'int16'
    INT8 = 'int8'
    FLOAT = 'float64'
    FLOAT64 = 'float64'
    FLOAT32 = 'float32'
    DATETIME64 = 'datetime64[ns]'
    TIMEDELTA64 = 'timedelta64[ns]'
    BOOLEAN = 'bool'
    CATEGORY = 'category'
    COMPLEX = 'complex'

def get_numeric_types():
    """
    Gets a list of numeric data types

    Returns:
    - list: List of numeric data types
    """
    return [
        DataTypes.INT8,
        DataTypes.INT16,
        DataTypes.INT32,
        DataTypes.INT64,
        DataTypes.FLOAT32,
        DataTypes.FLOAT64,
    ]